ehsm_kms_service/secret_manager_apis.js

```diff
@@ -0,0 +1,223 @@
+const crypto = require('crypto')
+const logger = require('./logger')
+const { v4: uuidv4 } = require('uuid')
+const {
+    napi_result,
+    _result,
+    base64_encode,
+    base64_decode
+} = require('./function')
+const {
+    cryptographic_apis,
+} = require('./apis')
+const ehsm_napi = require('./ehsm_napi')
+
+// get a string parameter from payload and base64 if you need
+function getParam_String(payload, key, needBase64) {
+    if (needBase64 == undefined) {
+        needBase64 = true;
+    }
+    let val = payload[key]
+    if (val != undefined) {
+        val = String(val)
+        if (needBase64) {
+            val = base64_encode(val)
+        }
+    }
+    return val
+}
+
+//check param is string and required
+function checkStringParam(param, required) {
+    if (param == '' || param == undefined) {
+        if (required) {
+            return false
+        } else {
+            return true
+        }
+    } else {
+        if (typeof (param) == 'string') {
+            return true
+        } else {
+            return false
+        }
+    }
+}
+
+function check_encryptionKeyId_format(encryptionKeyId) {
+    if (encryptionKeyId != '' && encryptionKeyId != undefined) {
```

Comment:
if encryptionKeyId is null, will this return false?

Reply:
Null will not appear, and it will be converted to string on line 22